domain/tutoringModel/tyche.py

```diff
@@ -0,0 +1,303 @@
+        elif dim4 == "glo":
+            lstyle[3] = "global"
+        else:
+            print("Invalid value!")
+        return lstyle
+
+    def get_probabilities(self, lstyle):
+        # Get initial state probability:
+
+        a, b, c, d = 0, 0, 0, 0
+        # Get further states probabilities:
+
+        e, f, g, h = 0, 0, 0, 0  # Generic
+        i, j, k, m = 0, 0, 0, 0  # Static
+        if "active" in lstyle:
+            a = np.array(t_config.Active_ini)
+            e = np.array(t_config.Active_g)
+            i = np.array(t_config.Active_static)
+        elif "reflective" in lstyle:
+            a = np.array(t_config.Reflective_ini)
+            e = np.array(t_config.Reflective_g)
+            i = np.array(t_config.Reflective_static)
+        if "visual" in lstyle:
+            b = np.array(t_config.Visual_ini)
+            f = np.array(t_config.Visual_g)
+            j = np.array(t_config.Visual_static)
+        elif "verbal" in lstyle:
+            b = np.array(t_config.Verbal_ini)
+            f = np.array(t_config.Verbal_g)
+            j = np.array(t_config.Verbal_static)
+        if "intuitive" in lstyle:
+            c = np.array(t_config.Intuitive_ini)
+            g = np.array(t_config.Intuitive_g)
+            k = np.array(t_config.Intuitive_static)
+        elif "sensing" in lstyle:
+            c = np.array(t_config.Sensing_ini)
+            g = np.array(t_config.Sensing_g)
+            k = np.array(t_config.Sensing_static)
+        if "sequential" in lstyle:
+            d = np.array(t_config.Sequential_ini)
+            h = np.array(t_config.Sequential_g)
+            m = np.array(t_config.Sequential_static)
+        elif "global" in lstyle:
+            d = np.array(t_config.Global_ini)
+            h = np.array(t_config.Global_g)
+            m = np.array(t_config.Global_static)
+        # Arithmetic means:
+
+        init_config = np.add(a, b)
+        init_config = np.add(init_config, c)
+        init_config = np.add(init_config, d)
+        init_config = np.divide(init_config, 4.0)
+
+        g_config = np.add(e, f)
+        g_config = np.add(g_config, g)
+        g_config = np.add(g_config, h)
+        g_config = np.divide(g_config, 4.0)
+
+        s_config = np.add(i, j)
+        s_config = np.add(s_config, k)
+        s_config = np.add(s_config, m)
+        s_config = np.divide(s_config, 4.0)
+
+        self.init_probabilities = init_config
+        self.probabilities = g_config
+
+    def get_learning_path_as_str(self):
+        """Convert the list of learning path into sting"""
+        str_learning_path = ""
+        contain_le = False
+        for ele in self.learningpath:
+            str_learning_path = str_learning_path + ele + ", "
+            contain_le = True
+        if contain_le:
+            str_learning_path = str_learning_path[:-2]
+        return str_learning_path
+
+    def get_learning_path(
+        self, input_learning_style={}, input_learning_element=[], last_element=False
+    ):

```

Comment:
```suggestion
        self, input_learning_style=None, input_learning_element=None, last_element=False
```

Reply:
{} and [] are changed to None.
default value of last_element is also changed to None.